src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockBuilder.cs

```diff
@@ -40,12 +40,14 @@ public TagHelperBlockBuilder(TagHelperBlock original)
         public TagHelperBlockBuilder(string tagName,
                                      SourceLocation start,
                                      IDictionary<string, SyntaxTreeNode> attributes,
+                                     bool selfClosing, 
```

Comment:
this is the pattern you should apply everywhere.  the only issue in `TagHelperBlockBuilder` is that `SelfClosing` should not have a setter.